plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,866 @@
+                "templateId": mso_template.template.get("templateId"),
+                "schemaId": mso_template.template.get("schemaId"),
+                "llfc": {},
+                "stp": {},
+                "l2Interface": {},
+                "lldp": {},
+                "linkLevel": {},
+                "mcp": {},
+                "portChannelPolicy": {},
+            }
+
+            if description:
+                payload["description"] = description
+
+            if domains:
+                domain_uuid = validate_domains(mso, domains, template, template_info)
+                payload["domains"] = domain_uuid
+
+            if synce:
+                existing_sync_e = validate_sync_e(mso, synce, template, template_info)
+                if synce in existing_sync_e:
+                    payload["syncEthPolicy"] = existing_sync_e[synce]
+
+            if access_macsec_policy:
+                existing_access_macsec_policy = validate_macsec_policy(mso, access_macsec_policy, template, template_info)
+                if access_macsec_policy in existing_access_macsec_policy:
+                    payload["accessMACsecPolicy"] = existing_access_macsec_policy[access_macsec_policy]
+
+            if cdp_admin_state:
+                payload["cdp"] = {"adminState": cdp_admin_state}
+
+            if pfc_admin_state:
+                payload["pfc"] = {"adminState": pfc_admin_state}
+
+            if llfc_transmit_state:
+                payload["llfc"]["transmitState"] = llfc_transmit_state
+            if llfc_receive_state:
+                payload["llfc"]["receiveState"] = llfc_receive_state
+
+            if stp_bpdu_filter:
+                payload["stp"]["bpduFilterEnabled"] = stp_bpdu_filter
+            if stp_bpdu_guard:
+                payload["stp"]["bpduGuardEnabled"] = stp_bpdu_guard
+
+            if l2_interface_qinq:
+                payload["l2Interface"]["qinq"] = l2_interface_qinq
+            if l2_interface_reflective_relay:
+                payload["l2Interface"]["reflectiveRelay"] = l2_interface_reflective_relay
+            if vlan_scope:
+                payload["l2Interface"]["vlanScope"] = vlan_scope
+
+            if lldp:
+                validate_lldp(mso, lldp)
+                payload["lldp"]["receiveState"] = lldp["receive_state"]
+                payload["lldp"]["transmitState"] = lldp["transmit_state"]
+
+            if link_level_debounce_interval:
+                payload["linkLevel"]["debounceInterval"] = link_level_debounce_interval
+            if link_level_bring_up_delay:
+                payload["linkLevel"]["bringUpDelay"] = link_level_bring_up_delay
+            if link_level_fec:
+                payload["linkLevel"]["fec"] = link_level_fec
+            if speed:
+                payload["linkLevel"]["speed"] = speed
+            if auto_negotiation:
+                payload["linkLevel"]["autoNegotiation"] = auto_negotiation
+
+            if mcp:
+                if mcp["admin_state"]:
```

Comment:
add a test case with false scenario

Reply:
A test case for false scenario exists as discussed.

Reply:
I do not see where this scenario exists. Because in your tests I do not see a test that specifies mcp configuration during create without `admin_state` config specified. In this case the `if mcp["admin_state"]` would be false.

Reply:
What was the action taken, because I still see this as the same when running tests?

Reply:
Added an additional case, just for coverage.